src/mapping/securitisationShelf.ts

```diff
@@ -0,0 +1,440 @@
+  pool!.lateFeeRepaid = securitisationShelfContract.totalLateFeePaid();
+  pool!.totalTrancheBalance = currencyContract
+    .balanceOf(Address.fromBytes(poolAddresses!.seniorTranche))
+    .plus(
+      currencyContract.balanceOf(
+        Address.fromBytes(poolAddresses!.juniorTranche)
+      )
+    );
+  pool!.trancheSupplyMaxBalance = operator
+    .totalDepositCurrencyJunior()
+    .plus(operator.totalDepositCurrencySenior());
+  pool!.prepaymentAbsorbedAmount =
+    securitisationShelfContract.prepaymentAbsorbedAmount();
+
+  // SecuritisationShelf-specific fields
+  pool!.outstandingShortfallInterestAmount =
+    securitisationShelfContract.outstandingShortfallInterestAmount();
+  pool!.outstandingShortfallPrincipalAmount =
+    securitisationShelfContract.outstandingShortfallPrincipalAmount();
+  pool!.servicerFeePaid = securitisationShelfContract.servicerFeePaid();
+  // Update reserve balance and EIS balance
+  let reserveContract = SecuritisationReserve.bind(
+    Address.fromBytes(poolAddresses!.reserve)
+  );
+  pool!.reserveBalance = reserveContract.balance();
+  pool!.eisBalance = reserveContract.eisBalance();
+
+  pool!.save();
+
+  // Update Tranche entities with SecuritisationTranche-specific fields
+  let seniorTranche = TrancheEntity.load(poolAddresses!.seniorTranche);
+  let juniorTranche = TrancheEntity.load(poolAddresses!.juniorTranche);
+
+  if (seniorTranche) {
+    seniorTranche.principalRepaid = seniorContract.principalRepaid();
+    seniorTranche.interestRepaid = seniorContract.interestRepaid();
+    seniorTranche.overduePrincipalAmount =
+      seniorContract.overduePrincipalAmount();
+    seniorTranche.lastRepaidTimestamp = seniorContract.lastRepaidTimestamp();
+    seniorTranche.totalDaysRepaid = seniorContract.totalDaysRepaid();
+    seniorTranche.totalRepaid = seniorTranche.principalRepaid!.plus(
+      seniorTranche.interestRepaid!
+    );
+    seniorTranche.save();
+  }
+
+  if (juniorTranche) {
+    juniorTranche.principalRepaid = juniorContract.principalRepaid();
+    juniorTranche.interestRepaid = juniorContract.interestRepaid();
+    juniorTranche.overduePrincipalAmount =
+      juniorContract.overduePrincipalAmount();
+    juniorTranche.lastRepaidTimestamp = juniorContract.lastRepaidTimestamp();
+    juniorTranche.totalDaysRepaid = juniorContract.totalDaysRepaid();
+    juniorTranche.totalRepaid = juniorTranche.principalRepaid!.plus(
+      juniorTranche.interestRepaid!
+    );
+    juniorTranche.save();
+  }
+
+  createRepayTransaction(event);
+}
+
+export function handleOriginatorFeePaidSecuritisationShelf(
+  event: OriginatorFeePaidSecuritisationShelf
+): void {
+  let pool = getPool(event.params.poolId);
+  pool!.originatorFeePaid = event.params.amount;
+  pool!.save();
+}
+
+export function getPool(poolId: BigInt): Pool | null {
+  let pool = Pool.load(getPoolId(poolId));
+  if (pool == null) {
+    log.error("Pool not found for ID: {}", [poolId.toHexString()]);
```

Comment:
use poolId.toString() instead, otherwise converting bigint to hex string will break